SCIP update patch

harrycheon opened the pull request.

harrycheon committed:
SCIP update patch

harrycheon committed:
tests and dependencies update

harrycheon committed:
fix: remove rpy2 dependency from default

- change github actions to only install default dependencies

harrycheon committed:
ci: environment lockfile update

bogdan-kulynych commented on quick.py:
Should this be in an examples/ folder?

harrycheon replied:
I think this my toy/draft file that leaked into branch. Can create an examples folder or delete

bogdan-kulynych commented on reachml/ext/training.py:
I guess all these ext files could be under research/

harrycheon replied:
Fair

bogdan-kulynych commented on reachml/reachable_set.py:
```diff
@@ -290,6 +293,124 @@ def add(
+
+        return len(vals)
+
+    def calculate_n(self, alpha, **kwargs):
```
Should we numerically compute this using the latest expression?

harrycheon replied:
Yes, we should update this function to match the paper

harrycheon committed:
chore: reorgnize research code

- Moved quick.py to research/iclr2025/examples/
- Moved reachml/ext to research/neurips2025/src

harrycheon committed:
chore: added todo for sample size calculation

- linked to GitHub issue #4

harrycheon committed:
fix: reachml/utils imports

- no longer import `train_model`
- deleted `check_processing_loss` function
  - in `src/utils.py` under research subfolders
